e2e/nomostest/gitproviders/repository.go

```diff
@@ -621,6 +621,15 @@ func (g *Repository) RenameBranch(current, new string) error {
 	return g.Push(current, "--delete")
 }
 
+// CurrentBranch returns the name of the current branch
+func (g *Repository) CurrentBranch() (string, error) {
+	out, err := g.Git("rev-parse", "--abbrev-ref", "HEAD")
```

Comment:
This wont work if on detached HEAD. Do we ever do that? Should we maybe comment it as a gotcha?

Reply:
The `Repository` interface currently does not support entering a detached HEAD state, so we never do this currently. I can add a comment